### PR TITLE
Add pop, truncate, and remove functions to ArrayString. 

### LINKED ISSUE #66: ArrayString lacks removal features
`ArrayString` lacks any removal features. `pop` `truncate` `remove` etc would be nice.

All of these may be copied from `std::String` AFAIK.

I will implement these and more if requested. I've opened this issue to discuss because this seems too obvious to have been accidentally omitted?

### PR DESCRIPTION
**Pull request by DenialAdams**
Fixes #66

This seemed "too easy" as I basically copied stuff from libstd (with tweaks to the comments and slight tweaks to the code, noted in truncate.) So please let me know if I did anything wrong!

### CODE REVIEW

**Comment by tbu-** [comment]
LGTM

**Comment by bluss** [comment]
Thanks a lot -- now we're just wondering why travis is not testing this.

File: src/array_string.rs [hunk 1]
```diff
@@ -220,6 +220,99 @@ impl<A: Array<Item=u8>> ArrayString<A> {
...
+    ///
+    /// assert_eq!(s.remove(0), 'f');
+    /// assert_eq!(s.remove(1), 'o');
+    /// assert_eq!(s.remove(0), 'o');
+    /// ```
+    #[inline]
+    pub fn remove(&mut self, idx: usize) -> char {
+        let ch = match self[idx..].chars().next() {
+            Some(ch) => ch,
+            None => panic!("cannot remove a char from the end of a string"),
```
Comment (bluss) [inline src/array_string.rs]: This is apparently what `String` says too. :+1:

**Comment by bluss** [comment]
@DenialAdams We'd like to have travis run before merging, since travis handles compatibility checks etc. Could you refresh the commit and force push or something, if it doesn't start? Like you'd amend the single commit and force push to the PR's branch.

**Comment by tbu-** [comment]
https://travis-ci.org/bluss/arrayvec/pull_requests doesn't show this pull request... Probably an issue on travis side.

**Comment by DenialAdams** [comment]
@bluss just did exactly what you said, hopefully this helps

**Comment by tbu-** [comment]
Hm....

**Comment by bluss** [comment]
Oh it's in the abuse list for some reason: https://travis-ci.org/bluss/arrayvec/requests

**Comment by DenialAdams** [comment]
> Oh it's in the abuse list for some reason

:( no idea why that would be. Looks like I have to email them, I'll do that

**Comment by tbu-** [comment]
Temporary fix: https://travis-ci.org/bluss/arrayvec/builds/292829811.

**Comment by tbu-** [comment]
Tests passed (see previous comment).

**Comment by DenialAdams** [comment]
Sorry about the extra force pushes, travis got back to me and asked me to test something (didn't work.) No code changed.

Thanks for running those tests @tbu- !

**Comment by bluss** [comment]
Thanks for these methods! Let's just merge, since @tbu- has made sure that the tests have run.

Travis support people seem really nice, I emailed them too.

**Pull request merged**